### PR TITLE
DAOS-8896 test: ior/hard_rebuild tests failed dfs_write() failed (28)

### PR DESCRIPTION
**Pull request by dinghwah**
Description: To increase timeout and ior segment size to 800000 per Di's suggestion

Quick-Functional: true
Test-tag: ec_ior_hard_online_rebuild
Signed-off-by: Ding Ho ding-hwa.ho@intel.com

### Before requesting gatekeeper:

* [ ] Two review approvals and any prior change requests have been resolved.
* [ ] Testing is complete and all tests passed or there is a reason documented in the PR why it should be force landed and forced-landing tag is set.
* [ ] `Features:` (or `Test-tag*`) commit pragma was used or there is a reason documented that there are no appropriate tags for this PR.
* [ ] Commit messages follows the guidelines outlined [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments).
* [ ] Any tests skipped by the ticket being addressed have been run and passed in the PR.

### Gatekeeper:

* [ ] You are the appropriate gatekeeper to be landing the patch.
* [ ] The PR has 2 reviews by people familiar with the code, including appropriate watchers.
* [ ] Githooks were used. If not, request that user install them and check copyright dates.
* [ ] Checkpatch issues are resolved.  Pay particular attention to ones that will show up on future PRs.
* [ ] All builds have passed.  Check non-required builds for any new compiler warnings.
* [ ] Sufficient testing is done. Check feature pragmas and test tags and that tests skipped for the ticket are run and now pass with the changes.
* [ ] If applicable, the PR has addressed any potential version compatibility issues.
* [ ] Check the target branch.   If it is master branch, should the PR go to a feature branch?  If it is a release branch, does it have merge approval in the JIRA ticket.
* [ ] Extra checks if forced landing is requested
  * [ ] Review comments are sufficiently resolved, particularly by prior reviewers that requested changes.
  * [ ] No new NLT or valgrind warnings.  Check the classic view.
  * [ ] Quick-build or Quick-functional is not used.
* [ ] Fix the commit message upon landing. Check the standard [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments). Edit it to create a single commit. If necessary, ask submitter for a new summary.

### CODE REVIEW

**Review by shimizukko — DISMISSED** [review]

**Review by daltonbohning — CHANGES_REQUESTED** [review]
Reducing `segment_count` doesn't really have an affect here, because IOR is ran with stonewall `-D 150`. Even with a reduced `segment_count`, it's very unlikely that processes write 800k segments each. IOR runs for 150s before stopping.
In fact, from a recent CI run:
https://build.hpdd.intel.com/job/daos-stack/job/daos/job/weekly-testing/303/artifact/Functional%20Hardware%20Large/ior/hard_rebuild.py/job.log
`Stat() of aggregate file size      = 120969259008`
But from this PR:
https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-12189/3/artifact/Functional%20Hardware%20Large/ior/hard_rebuild.py/repeat001/job.log
`Stat() of aggregate file size      = 208935141376`
So this PR actually generated a larger file!

The proper fix here is to reduce `-D 150` as necessary. I'm not sure what that should be. Maybe 120? 90?

**Review by daltonbohning — APPROVED** [review]

**Review by phender — DISMISSED** [review]

File: src/tests/ftest/ior/hard_rebuild.yaml [hunk 1]
```diff
@@ -7,7 +7,7 @@ hosts:
     12_server:
       test_servers: server-[1-6]
   test_clients: 2
-timeout: 1000
+timeout: 1600
```
Comment (mchaarawi) [inline src/tests/ftest/ior/hard_rebuild.yaml:10]: it sounds weird that we increase the timeout and decrease the stone wall.
is the issue a performance regression?

**Review by mchaarawi — CHANGES_REQUESTED** [review]
i reviewed the comments in the ticket. it sounds to me that the change required is just the stonewall and not the test timeout. please correct me if im wrong.

**Comment by daltonbohning** [comment]
> i reviewed the comments in the ticket. it sounds to me that the change required is just the stonewall and not the test timeout. please correct me if im wrong.

@mchaarawi I think you're right. The misunderstanding about segments in relation to stonewall is probably why it was suggested to increase the timeout.

From this run, the test completed in under 5 minutes each of 10 times:
https://build.hpdd.intel.com/blue/organizations/jenkins/daos-stack%2Fdaos/detail/PR-12189/4/tests

**Review by shimizukko — APPROVED** [review]

**Review by phender — APPROVED** [review]

**Pull request merged**